### PR TITLE
CLB PoC

### PR DESCRIPTION
**Pull request by euonymos**
This PR is a temporary thing until we move to the CLB repository. Currently, it's an attempt to replace PSM with an in-place emulator (Clb).

Some notes:
* I removed `Run` being exposed in the users's interface (some changes have to do with it)

### CODE REVIEW

File: src/GeniusYield/Clb/Clb.hs [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+
+-- | Emulator state: ledger state + some additional things
+-- FIXME: remove non-state parts like MockConfig and Log (?)
+data ClbState = ClbState
+  { emulatedLedgerState :: !EmulatedLedgerState
+  , mockConfig          :: !MockConfig
+  , mockDatums          :: !(Map P.DatumHash P.Datum)
+  , mockInfo            :: !(Log LogEntry)
+  , mockFails           :: !(Log FailReason)
+  , mockCurrentSlot     :: !Slot -- FIXME: sync up with emulatedLedgerState
```
Comment (GeorgeFlerovsky) [inline src/GeniusYield/Clb/Clb.hs:105]: I don't really like the terminology "mock" and "emulated" in reference to the CLB ledger — CLB runs a real Cardano ledger.

The only things it emulates are time, network, and consensus — these can be called emulated/mock.

File: src/GeniusYield/Clb/Clb.hs [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+    genesisTxId :: L.TxId L.StandardCrypto
+    genesisTxId = L.TxId $ L.unsafeMakeSafeHash dummyHash
+
+    -- Hash for genesis transaction
+    dummyHash :: Crypto.Hash Crypto.Blake2b_256 Core.EraIndependentTxBody
+    dummyHash = Crypto.castHash $ Crypto.hashWith CBOR.serialize' ()
+
+--------------------------------------------------------------------------------
+-- Trace log (from PSM)
+--------------------------------------------------------------------------------
```
Comment (GeorgeFlerovsky) [inline src/GeniusYield/Clb/Clb.hs:188]: Logging and pretty printing should be in a separate module

Comment (GeorgeFlerovsky) [inline src/GeniusYield/Clb/Clb.hs:188]: Logging and pretty printing should be in a separate module

File: src/GeniusYield/Clb/Params.hs [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+babbageOnly _ = error "Sorry, Babbage only"
+
+-- Alonzo
+
+-- | Default Alonzo era parameters
+defaultAlonzoParams :: PParams
+defaultAlonzoParams = AlonzoParams defaultAlonzoParams'
+
+defaultAlonzoParams' :: L.PParams (L.AlonzoEra L.StandardCrypto)
+defaultAlonzoParams' =
```
Comment (GeorgeFlerovsky) [inline src/GeniusYield/Clb/Params.hs:43]: I dislike hardcoding config values in code. Is there a way to get these from a config file?

**Review by GeorgeFlerovsky — APPROVED** [review]
On general inspection, looks good. 

I'm not convinced that you need to import GY Atlas or that CLB needs to live in that repo, but it's OK for now.

File: src/GeniusYield/Clb/Clb.hs (hunk 1 above)
Comment (uhbif19) [reply src/GeniusYield/Clb/Clb.hs:105]: `mockCurrentSlot` - seem to duplicate one in `EmulatedLedgerState`.

**Comment by euonymos** [comment]
We moved the code of CLB into its own repo [here](https://github.com/mlabs-haskell/clb).

**Pull request closed without merging**